tests/e2e/iothub_e2e/utils.py

```diff
@@ -27,60 +27,45 @@ def get_fault_injection_message(fault_injection_type):
     return fault_message
 
 
-def create_client_object(device_identity, client_kwargs, DeviceClass, ModuleClass):
-
-    if test_config.config.identity in [
-        test_config.IDENTITY_DEVICE,
-        test_config.IDENTITY_EDGE_LEAF_DEVICE,
-    ]:
-        ClientClass = DeviceClass
-    elif test_config.config.identity in [
-        test_config.IDENTITY_MODULE,
-        test_config.IDENTITY_EDGE_MODULE,
-    ]:
-        ClientClass = ModuleClass
-    else:
-        raise Exception("config.identity invalid")
+def create_session_object(device_identity, client_kwargs):
 
     if test_config.config.auth == test_config.AUTH_CONNECTION_STRING:
         logger.info(
-            "Creating {} using create_from_connection_string with kwargs={}".format(
-                ClientClass, client_kwargs
+            "Creating session using create_from_connection_string with kwargs={}".format(
+                client_kwargs
             )
         )
 
-        client = ClientClass.create_from_connection_string(
+        session = IoTHubSession.from_connection_string(
             device_identity.connection_string, **client_kwargs
         )
     elif test_config.config.auth == test_config.AUTH_SYMMETRIC_KEY:
         logger.info(
-            "Creating {} using create_from_symmetric_key with kwargs={}".format(
-                ClientClass, client_kwargs
-            )
+            "Creating session using create_from_symmetric_key with kwargs={}".format(client_kwargs)
         )
 
-        client = ClientClass.create_from_symmetric_key(
-            device_identity.primary_key,
-            test_env.IOTHUB_HOSTNAME,
-            device_identity.device_id,
+        session = IoTHubSession(
+            shared_access_key=device_identity.primary_key,
+            hostname=test_env.IOTHUB_HOSTNAME,
+            device_id=device_identity.device_id,
             **client_kwargs
         )
     elif test_config.config.auth == test_config.AUTH_SAS_TOKEN:
         logger.info(
-            "Creating {} using create_from_sastoken with kwargs={}".format(
-                ClientClass, client_kwargs
-            )
+            "Creating session using create_from_sastoken with kwargs={}".format(client_kwargs)
         )
 
-        client = ClientClass.create_from_sastoken(device_identity.sas_token, **client_kwargs)
+        # client = ClientClass.create_from_sastoken(device_identity.sas_token, **client_kwargs)
+
+        raise Exception("{} Auth not yet implemented".format(test_config.config.auth))
```

Comment:
Nit: this is the exact use case for `NotImplementedError`, and it's always safer to throw a specific error than a general one.